### ISSUE TITLE
Constant CPU usage.

### ISSUE THREAD

**Issue by aberfan** [JohnnyRS/PandaCrazy-Max#51]
PC now seems to use an almost constant CPU usage that varies between 6%-9.4% on my machine. This seems new. It's only on the most recent builds and despite being low is noticeable on my system. 

I don't think my timers are far off from default, with 1s for panda, 950 for search and 2s for queue watching. I don't use whatever the search specific thing is and just use the "classic" panda/searches.  I posted in another comment about weird search behavior where it just freezes.  Also an issue for me is that pandas spawned from search hits seemed to just stop. I don't know if this were some interaction with the so called _temporary_ pandas but now it's very confusing what all these do and the help doesn't help.

My desired behavior is when I have a panda which is created from a search, I want that panda to run until I turn it off, even if that's 6 hours.  There are many qualifications and surveys which can be caught 2 or 3 hours after they initially hit, since when I mentioned this before it seemed I needed to justify my work flow.  Can there be an option here to please just run the damn thing until I decide to turn it off and never sooner?

But it would have to actually create the panda in the first place which has seemed iffy in the last couple of versions.  Whatever changes made recently have had an adverse affect.

I do appreciate the extension and that it's free and a thankless labor but I also know you take pride in your work so if these things could be addressed it'd be awesome! Thank you!

**Comment by JohnnyRS** [comment]
Yes there is a mistake made when checking for ne HITs on the search page that is causing this. Working on a new update that will fix that issue. This also seems to be causing delays when turning panda's on and off which can cause a panda job to be stuck. Hope this will fix those problems as well. But I am planning on adding an error correction to find any panda jobs stuck.

The problem with running a panda created by a search for 6 hours is that it is inefficient use of panda jobs. A search job should be able to find that hit and run the panda job in 6 hours too. For example: If you have a panda created from a search job running for 6 hours and 2 other panda's running then that panda will only check around 3 seconds. Compare that to a search job will checks around every second. The search job should find that panda faster in that circumstance. The only way a panda job will be faster than a search job is if you only have that one panda running for 6 hours with no other panda's running then the panda job may find it before a search job. You can have hundreds of search jobs running at the same time and it will constantly be checking around every second. That is the reason for the searchUI page to make it easier to have multiple search jobs running at once. I'm not being a control freak here but trying to make things more efficient for users because it may be hard to understand how all this works.

The problems with panda jobs created by searches should be fixed now if the options are correct. The current version on github has some bad bugs so hoping to have it updated sometime today. If everything works I am hoping to update it on the chrome store.

**Comment by aberfan** [comment]
I pull directly from github so maybe that's why. I don't use the Chrome store and don't want to.

Thank you for fixing things and glad to hear you know what the problems are.

For the search thing, yes. I have one panda running without others.  I've caught things in the past where it never showed up in a search, but panda gets it. These are mostly quals or some academic things, such as Samuel Gershwin.  I understand the need to "make things more efficient for users" but it would be nice if you trust people know what they're doing and let them shoot themselves in the foot if they want.  

Please make it simple for people who just want a requester search to create a panda that runs until they tell it to stop, to do so.   This week the search not working and me not having noticed has had financial consequences. I wish I could chip in and help some way with the code.  

Thank you again for your hard work with it!

**Comment by JohnnyRS** [comment]
But the thing is there is already a way to do exactly what you want manually. If you see a panda job that you want to run forever then turn it off and turn it on manually then it should run forever or that is how it is supposed to run. There may be some bugs that was causing it to stop but I believe it's all fixed now. Be sure the panda job options for the stop collecting are set to 0. Also check the search jobs for those options too. I actually do that to some panda jobs myself so I can turn it off when I want to. I can make the max limit for panda's created from a search job higher but I would rather not put in an unlimited option for automated features.

**Comment by aberfan** [comment]
Unlimited option for automated features should exist; the compromise here would simply be a menu option that you'd enable, that says '0' means forever.  

I hear what you're saying, but there are definite cases where just an hour is too short. It's non-intuitive that there is an artificial limit. I recognize that my usage is much less than most people. I normally only do have one or two pandas on. If a search trigger creates a panda, a lot of the times it'll be for one that has a very long time.  If you have a 3 hour qualification or survey that doesn't mean it has to wait that long. Some people may return it after an hour and a half.  With the panda automatically off, and without it showing up in search again and re-triggering the panda, it means that unless you're constantly babysitting, you aren't going to get these.  I'd rather not have to constantly babysit.  Unless there's a technical reason this can't be done or you've heard feedback from others, at least having a way for people to leave pandas created by triggers until manually shut off would be really helpful.

But more helpful would just be having it so it even searches in the first place, so I'll stop bugging you while you're working on those bugs!

**Comment by JohnnyRS** [comment]
There's no technical reason I couldn't do it. But it goes against my number 1 rule of limits to automation and no limits to manually pressed stuff. I just updated this new version with the fixes. I changed the maximum temporary search panda duration to 6 hours from 2 hours but it's set in seconds. Hope that will do what you want. Also if you're not using the search page then be aware there are search general options on that page that control the search jobs too. Make sure those options are correct.
